docs/topics/gradle/gradle-compilation-and-caches.md

```diff
@@ -331,6 +348,34 @@ kotlin.build.report.http.password=somePassword
 kotlin.build.report.label=some_label
 ```
 
+### Limit of custom values
+
+To collect build scans' statistics, Kotlin build reports use [Gradle's custom values](https://docs.gradle.com/enterprise/tutorials/extending-build-scans/). 
+If you have a big project, a number of such custom values may be more than 1000 and these values consume a storage's space. 
+Other Gradle plugins write custom values too. 
+You can limit a maximum number of custom values in `gradle.properties` depending on your project's size:
+
+```properties
+kotlin.build.report.build_scan.custom_values_limit=1500
+```
+
+If this amount is exceeded, you can see a following message in logs:
+
+```
+Maximum number of custom values (1,500) exceeded
+```
+
```

Comment:
Or this variant:

To collect build scans' statistics, Kotlin build reports use [Gradle's custom values](https://docs.gradle.com/enterprise/tutorials/extending-build-scans/). 
If you have a big project, a number of such custom values may be quite big.

If number of custom values exceeds some value, you can see a following message in logs:

```
Maximum number of custom values (1,000) exceeded
```

To solve this, you can limit a maximum number of custom values in `gradle.properties` depending on your project's size:

```properties
kotlin.build.report.build_scan.custom_values_limit=1500
```

Keep in mind that these values consume a storage's space and other Gradle plugins write custom values too.

Reply:
Please choose one that you like more.